src/context.ts

```diff
@@ -19,40 +29,23 @@ export function buildPath(path: ResponsePath | undefined) {
   return segments.reverse().join(".");
 }
 
-export interface SpanContext extends Object {
-  _spans: Map<string, Span>;
-  getSpanByPath(info: ResponsePath): Span | undefined;
-  addSpan(span: Span, info: GraphQLResolveInfo): void;
-  // Passed in from the outside context
-  requestSpan?: Span;
-}
-
-function isSpanContext(obj: any): obj is SpanContext {
-  return (
-    obj.getSpanByPath instanceof Function && obj.addSpan instanceof Function
-  );
-}
-
-export function requestIsInstanceContextRequest<CTX extends SpanContext>(
-  request: GraphQLRequestContext<CTX | Object>
-): request is GraphQLRequestContext<CTX> {
-  return isSpanContext(request.context);
-}
-
-// TODO: think about using symbols to hide these
-export function addContextHelpers(obj: any): SpanContext {
-  if (isSpanContext(obj)) {
-    return obj;
+export function addContextHelpers<Context extends ContextForApolloOpenTracing>(
+  contextValue: Context
+): WithRequiredSpanContext<Context> {
+  if (hasSpanContext(contextValue)) {
+    return contextValue;
   }
 
-  obj._spans = new Map<string, Span>();
-  obj.getSpanByPath = function (path: ResponsePath): Span | undefined {
-    return this._spans.get(buildPath(isArrayPath(path) ? path.prev : path));
-  };
-
-  obj.addSpan = function (span: Span, info: GraphQLResolveInfo): void {
```

Comment:
Is `context.getSpanByPath` and `context.addSpan` supposed to be part of the public API provided by this library? 

I don't see it referenced in README.md, other than: `onFieldResolve(source: any, args: { [argName: string]: any }, context: SpanContext, info: GraphQLResolveInfo): Allow users to add extra information to the span` which implies either:
- you're supposed to be able to use these fields specifically when doing `onFieldResolve`, or;
- this is just a consequence of mutating the context object and we're just being honest

(I've interpreted the latter given `think about using symbols to hide these`, but lmk!)